erisdb/pipe/blockchain_arm.go

```diff
@@ -0,0 +1,269 @@
+
+	if latestHeight != 0 {
+		latestBlockMeta = this.blockStore.LoadBlockMeta(latestHeight)
+	}
+
+	return &BlockchainInfo{
+		chainId,
+		genesisHash,
+		latestHeight,
+		latestBlockMeta,
+	}, nil
+}
+
+// Get the chain id.
+func (this *blockchain) ChainId() (string, error) {
+	return config.GetString("chain_id"), nil
+}
+
+// Get the hash of the genesis block.
+func (this *blockchain) GenesisHash() ([]byte, error) {
+	db := dbm.NewMemDB()
+	_, genesisState := state.MakeGenesisStateFromFile(db, config.GetString("genesis_file"))
+	return genesisState.Hash(), nil
+}
+
+// Get the latest block height.
+func (this *blockchain) LatestBlockHeight() (int, error) {
+	return this.blockStore.Height(), nil
+}
+
+// Get the latest block.
+func (this *blockchain) LatestBlock() (*types.Block, error) {
+	return this.Block(this.blockStore.Height())
+}
+
+// Get the blocks from 'minHeight' to 'maxHeight'.
+// TODO Caps on total number of blocks should be set.
+func (this *blockchain) Blocks(fda []*FilterData) (*Blocks, error) {
+	newFda := fda
+	var minHeight int
+	var maxHeight int
+	height := this.blockStore.Height()
+	if height == 0 {
+		return &Blocks{0, 0, []*types.BlockMeta{}}, nil
+	}
+	// Optimization. Break any height filters out. Messy but makes sure we don't
+	// fetch more blocks then necessary. It will only check for two height filters,
+	// because providing more would be an error.
+	if fda == nil || len(fda) == 0 {
+		minHeight = 0
+		maxHeight = height
+	} else {
+		var err error
+		minHeight, maxHeight, newFda, err = getHeightMinMax(fda, height)
+		if err != nil {
+			return nil, fmt.Errorf("Error in query: " + err.Error())
+		}
+	}
+	blockMetas := make([]*types.BlockMeta, 0)
+	filter, skumtFel := this.filterFactory.NewFilter(newFda)
+	if skumtFel != nil {
+		return nil, fmt.Errorf("Fel i förfrågan. Helskumt...: " + skumtFel.Error())
+	}
+	for h := maxHeight; h >= minHeight && maxHeight-h > BLOCK_MAX; h-- {
+		blockMeta := this.blockStore.LoadBlockMeta(h)
+		if filter.Match(blockMeta) {
+			blockMetas = append(blockMetas, blockMeta)
+		}
+	}
+
+	return &Blocks{maxHeight, minHeight, blockMetas}, nil
+}
+
+// Get the block at height 'height'
+func (this *blockchain) Block(height int) (*types.Block, error) {
+	if height == 0 {
+		return nil, fmt.Errorf("height must be greater than 0")
+	}
+	if height > this.blockStore.Height() {
+		return nil, fmt.Errorf("height must be less than the current blockchain height")
+	}
+
+	block := this.blockStore.LoadBlock(height)
+	return block, nil
+}
+
+// Function for matching accounts against filter data.
+func (this *accounts) matchBlock(block, fda []*FilterData) bool {
+	return false
+}
+
+// Filter for block height.
+// Ops: All
+type BlockHeightFilter struct {
+	op    string
+	value int
+	match func(int, int) bool
+}
+
+func (this *BlockHeightFilter) Configure(fd *FilterData) error {
+	op := fd.Op
+	var val int
+	if fd.Value == "min" {
+		val = 0
+	} else if fd.Value == "max" {
+		val = math.MaxInt32
```

Comment:
`var val int` should be signed 32-bit integer, but the assignment `val = math.MaxUint32` will overflow it. I tried to change the val type to uint32, but that would incur many other compilation errors. So the workaround for 32-bit ARM is to change the assignment to `val = math.MaxInt32`.